samza-core/src/main/java/org/apache/samza/zk/ZkKeyBuilder.java

```diff
@@ -51,6 +51,7 @@
 
   static final String PROCESSORS_PATH = "processors";
   static final String JOBMODEL_GENERATION_PATH = "JobModelGeneration";
+  static final String JOB_MODEL_UPGRADE_BARRIER = "jobModelUpgradeBarrier";
 
   public ZkKeyBuilder(String pathPrefix) {
     if (Strings.isNullOrEmpty(pathPrefix)) {
```

Comment:
It is not related to your change, though i noticed we trim the `pathPrefix` before assignment. Ideally we should trim it before the check to make sure `"    "` isn't considered a valid prefix